src/main/java/ch/njol/skript/expressions/ExprInventorySlot.java

```diff
@@ -41,78 +32,87 @@
 import ch.njol.skript.util.slot.Slot;
 import ch.njol.util.Kleenean;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.PlayerInventory;
+import org.jetbrains.annotations.Nullable;
+
 @Name("Inventory Slot")
-@Description({"Represents a slot in an inventory. It can be used to change the item in an inventory too."})
-@Examples({"if slot 0 of player is air:",
+@Description("Represents a slot in an inventory. It can be used to change the item in an inventory too.")
+@Examples({
+	"if slot 0 of player is air:",
 	"\tset slot 0 of player to 2 stones",
 	"\tremove 1 stone from slot 0 of player",
-	"\tadd 2 stones to slot 0 of player",
-	"\tclear slot 1 of player"})
-@Since("2.2-dev24")
+	"\tadd 2 stones to slot 0 of inventory of all players",
+	"\tclear slot 1 of player"
+})
+@Since("2.2-dev24, INSERT VERSION (multiple inventories)")
 public class ExprInventorySlot extends SimpleExpression<Slot> {
 	
 	static {
 		Skript.registerExpression(ExprInventorySlot.class, Slot.class, ExpressionType.COMBINED,
-				"[the] slot[s] %numbers% of %inventory%", "%inventory%'[s] slot[s] %numbers%");
+				"[the] slot[s] %numbers% of %inventories%", "%inventories%'[s] slot[s] %numbers%");
 	}
 
 	@SuppressWarnings("null")
 	private Expression<Number> slots;
 	@SuppressWarnings("null")
-	private Expression<Inventory> invis;
-	
-	@SuppressWarnings({"null", "unchecked"})
+	private Expression<Inventory> inventories;
+
 	@Override
+	@SuppressWarnings({"null", "unchecked"})
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		if (matchedPattern == 0){
 			 slots = (Expression<Number>) exprs[0];
-			 invis = (Expression<Inventory>) exprs[1];
+			 inventories = (Expression<Inventory>) exprs[1];
 		} else {
 			 slots = (Expression<Number>) exprs[1];
-			 invis = (Expression<Inventory>) exprs[0];			
+			 inventories = (Expression<Inventory>) exprs[0];
 		}
 		return true;
 	}
 
 	@Override
 	@Nullable
 	protected Slot[] get(Event event) {
-		Inventory invi = invis.getSingle(event);
-		if (invi == null)
-			return null;
-		
 		List<Slot> inventorySlots = new ArrayList<>();
-		for (Number slot : slots.getArray(event)) {
-			if (slot.intValue() >= 0 && slot.intValue() < invi.getSize()) {
+		Number[] slotsArray = slots.getArray(event);
```

Comment:
no point in having this as a variable when you can use it directly in the loop or use a stream as the return value since you're not checking the size of this anymore.